contract-sdk/types/src/crypto.rs

```diff
@@ -0,0 +1,73 @@
+use std::convert::TryFrom;
+
+use k256::{
+    ecdsa::{recoverable, Signature},
+    EncodedPoint,
+};
+use thiserror::Error;
+
+/// ECDSA recover error.
+#[derive(Error, Debug, cbor::Encode, cbor::Decode)]
+pub enum Error {
+    #[error("malformed input")]
+    MalformedInput,
+    #[error("malformed signature")]
+    MalformedSignature,
+    #[error("signature verification failed")]
+    VerificationFailed,
+}
+
+/// ECDSA public key recovery method.
+pub fn ecdsa_recover(input: &[u8]) -> Result<Vec<u8>, Error> {
+    // [32] hash, [32] r, [32] s, [1] v.
+    if input.len() != 97 {
+        return Err(Error::MalformedInput);
+    }
+
+    let mut msg = [0u8; 32];
+    let mut sig = [0u8; 64];
+    let v = input[96];
+
+    msg[0..32].copy_from_slice(&input[0..32]);
+    sig[0..].copy_from_slice(&input[32..96]);
+
+    let signature = Signature::try_from(&sig[..]).map_err(|_| Error::MalformedSignature)?;
```

Comment:
there's also a `from_scalars` if you'd prefer to have the r and s explicit

Reply:
Thanks, switched to using `from_scalars`